src/pd_gmres.m

```diff
@@ -292,62 +292,43 @@
         end
 
         mIteration(iter(size(iter, 1), :) + 1, 1) = m;
-        v = zeros(n, m + 1);
-        w = zeros(n, m);
+
+        % Compute normalized residual vector
         r = b - A * xInitial;
         beta = norm(r);
-        v(:, 1) = r / beta;
-        h = zeros(m + 1, m);
+        v1 = r / beta;
+
+        % Modifed Gram-Schmidt Arnoldi iteration
+        [H, V] = modified_gram_schmidt_arnoldi(A, v1, m);
 
-        % Modified Gram Schmidt-Arnoldi
-        for j = 1:m
-            w(:, j) = A * v(:, j);
-            for i = 1:j
-                h(i, j) = w(:, j)' * v(:, i);
-                w(:, j) = w(:, j) - h(i, j) * v(:, i);
-            end
-            h(j + 1, j) = norm(w(:, j));
-            if h(j + 1, j) == 0
-                m = j;
-                h2 = zeros(m + 1, m); % Comment by JCC: VERIFY!!! (why?)
-                for k = 1:m
-                    h2(:, k) = h(:, k);
-                end
-                h = h2;
-            else
-                v(:, j + 1) = w(:, j) / h(j + 1, j);
-            end
-        end
         g = zeros(m + 1, 1);
         g(1, 1) = beta;
 
         % Plane rotations (QR decompostion)
```

Comment:
Be careful with implementation with Givens rotations for complex systems (See Saad book, page 193).

Reply:
Will do!